test/optimizer/test_qgt_logic.py

```diff
@@ -391,11 +403,39 @@ def test_scale_invariant_regularization(e, outdtype, pardtype):
         centered_oks, e.samples
     )
 
-    centered_oks_scaled, scale = qgt_jacobian_pytree_logic._rescale(centered_oks)
+    centered_oks_scaled, scale = qgt_jacobian_pytree_logic._rescale(
+        centered_oks, offset
+    )
     actual = mv(e.v, centered_oks_scaled)
     expected = reassemble_complex(e.S_real_scaled @ e.v_real_flat, target=e.target)
     assert tree_allclose(actual, expected)
     tree_samedtypes(actual, expected)
 
 
+@pytest.mark.parametrize(
+    "inputs, expected",
+    [
+        ((None, None, None), (0.01, 0.0)),
+        ((0.02, None, None), (0.02, 0.0)),
+        ((None, 0.02, None), (0.0, 0.02)),
+        ((0.01, 0.02, None), (0.01, 0.02)),
+        ((None, None, False), (0.01, 0.0)),
+        ((0.02, None, False), (0.02, 0.0)),
+        ((None, 0.02, False), "error"),
+        ((0.01, 0.02, False), "error"),
+        ((None, None, True), (0.0, 0.01)),
+        ((0.02, None, True), (0.0, 0.02)),
+        ((None, 0.02, True), "error"),
+        ((0.01, 0.02, True), "error"),
```

Comment:
```suggestion
        ((0.01, 0.02, True), "error"),
        ((None, None, 0.0), "error"),
```
because codecov was complaining